Modules/_peg_parser.c

```diff
@@ -80,12 +80,8 @@ _Py_parse_string(PyObject *self, PyObject *args, PyObject *kwds)
     flags.cf_flags = PyCF_IGNORE_COOKIE;
 
     mod_ty res;
-    if (oldparser) {
-        res = PyParser_ASTFromString(the_string, "<string>", mode, &flags, arena);
-    }
-    else {
-        res = PyPegen_ASTFromString(the_string, mode, &flags, arena);
-    }
+    res = (oldparser ? PyParser_ASTFromString : PyPegen_ASTFromString)
+        (the_string, "<string>", mode, &flags, arena);
```

Comment:
Although I like the shortness of this, it took me a second to parse this as a call....could we move this to an if-else block or maybe capture the function in a different variable?

Reply:
Done!